### ISSUE TITLE
Apps wants to downgrade Firefox Focus

### ISSUE THREAD

**Issue by RomainL972** [Tobi823/ffupdater#9]
I receive notifications for ffupdater browser updates but when I go on the app it wants to downgrade Firefox Focus
![Screenshot_20200705-222709](https://user-images.githubusercontent.com/24463530/86541634-b6783500-bf0e-11ea-816f-798471689274.jpg)

**Comment by Tobi823** [comment]
That's strange - can you try it again with the latest FFUpdater version 68.3.7?

**Comment by RomainL972** [comment]
I already have to 68.3.7 version.

**Comment by Tobi823** [comment]
Are you sure? In 68.3.7 Firefox Focus is downloaded from the Mozilla CI Server which uses timestamps (instead of versions).

![Screenshot_20200706-221605](https://user-images.githubusercontent.com/10066140/86638143-10443200-bfd7-11ea-83a3-640a0f2f3c03.png)

Can you:
- update the F-Droid repository by downswiping in the F-Droid app
- delete FFUpdater and reinstall it (to force the latest FFUpdater-version and delete all cached data)
- install Firefox Focus and check if there is still a bug?

**Comment by Tobi823** [comment]
I think I've fixed the issue in version 69.0.0 (this version will be available in few days on F-Droid)

Feel free to reopen the ticket if the bug is still present in 69.0.0

**Comment by TPS** [comment]
This is again a problem:

<Details>

<Summary>

Main Screen

</Summary>

![FfU 0 Home.18Direct.png](https://user-images.githubusercontent.com/185902/218310296-de2aebd7-9c89-457b-8f2c-041909c4b0a9.png)

</Details>

<Details>

<Summary>

Install Screen

</Summary>

![FfU 1 Downgrade Error.4Direct.png](https://user-images.githubusercontent.com/185902/218310574-cea68621-98e2-464a-bcc5-9d321983e880.png)

</Details

<Details>

<Summary>

Installed (updated from GPlay over previous FFUpdated version) Ff Focus Info

</Summary>

![Ff 110.0.9Direct.png](https://user-images.githubusercontent.com/185902/218310653-cd3fcc48-a66f-4df2-beca-d7b115eb7a9c.png)

</Details>

<Details>

<Summary>

Downloaded Ff Focus info

</Summary>

![Ff 109.2.0.8Direct.png](https://user-images.githubusercontent.com/185902/218310662-935500bb-d57a-4dc4-88bd-3ebf04d5f86f.png)


<Details>

<Summary>

Stacktrace:

</Summary>

```
java.lang.RuntimeException: Failed to install FIREFOX_FOCUS in the foreground.
	at de.marmaro.krt.ffupdater.DownloadActivity.installApp(SourceFile:177)
	at de.marmaro.krt.ffupdater.DownloadActivity.access$installApp(Unknown Source:0)
	at de.marmaro.krt.ffupdater.DownloadActivity$installApp$1.invokeSuspend(Unknown Source:11)
	at w3.a.resumeWith(Unknown Source:8)
	at i4.l0.run(Unknown Source:102)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:233)
	at android.os.Looper.loop(Looper.java:344)
	at android.app.ActivityThread.main(ActivityThread.java:8210)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034)
Caused by: de.marmaro.krt.ffupdater.installer.exceptions.InstallationFailedException: INSTALL_FAILED_VERSION_DOWNGRADE: Package Verification Result. The installation failed because one or more of the APKs was invalid.
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.fail(Unknown Source:4)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.handleAppInstallationResult(Unknown Source:31)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.access$handleAppInstallationResult(Unknown Source:0)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller$registerIntentReceiver$1.onReceive(Unknown Source:2)
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1751)
	at android.app.LoadedApk$ReceiverDispatcher$Args$$ExternalSyntheticLambda0.run(Unknown Source:2)
	... 8 more
```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.5 (131) release |
| Device | DE2118 (OnePlusN200TMO, OnePlusN200TMO, holi) |
| Manufacturer | OnePlus (OnePlus) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 12 (SDK: 31) |
| OS | dg02-pool03-kvm8, root, release-keys, 1670228878000 |

</Details>

**Comment by TPS** [comment]
Again a problem w/ latest Ff Focus

<details>

Stacktrace:
```
java.lang.RuntimeException: Failed to install FIREFOX_FOCUS in the foreground.
	at de.marmaro.krt.ffupdater.DownloadActivity.installApp(DownloadActivity.kt:405)
	at de.marmaro.krt.ffupdater.DownloadActivity.access$installApp(DownloadActivity.kt:59)
	at de.marmaro.krt.ffupdater.DownloadActivity$installApp$1.invokeSuspend(DownloadActivity.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:233)
	at android.os.Looper.loop(Looper.java:344)
	at android.app.ActivityThread.main(ActivityThread.java:8210)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034)
Caused by: de.marmaro.krt.ffupdater.installer.exceptions.InstallationFailedException: INSTALL_FAILED_VERSION_DOWNGRADE: Package Verification Result. The installation failed because one or more of the APKs was invalid.
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.fail(SessionInstaller.kt:187)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.handleAppInstallationResult(SessionInstaller.kt:126)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller.access$handleAppInstallationResult(SessionInstaller.kt:29)
	at de.marmaro.krt.ffupdater.installer.impl.SessionInstaller$registerIntentReceiver$1.onReceive(SessionInstaller.kt:207)
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1751)
	at android.app.LoadedApk$ReceiverDispatcher$Args$$ExternalSyntheticLambda0.run(Unknown Source:2)
	... 8 more
```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 77.7.10 (136) release |
| Device | DE2118 (OnePlusN200TMO, OnePlusN200TMO, holi) |
| Manufacturer | OnePlus (OnePlus) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 12 (SDK: 31) |
| OS | dg02-pool03-kvm216, root, release-keys, 1676459476000 |         

</details>

**Comment by Tobi823** [comment]
The current version for Focus is 110.1.0 (see https://github.com/mozilla-mobile/firefox-android/releases/latest and  https://www.apkmirror.com/apk/mozilla/firefox-focus-private-browser)

I suspect that the maintainers of https://github.com/mozilla-mobile/firefox-android accidentally released an update with a wrong version (or a beta update as a normal update). And this temporary version was installed by FFUpdater on your phone.

AFAIK there is no bug in FFUpdater. I guess you have to wait for the release of the real 111.0 version.

**Comment by TPS** [comment]
Actually, v111.0 downloaded & installed from GPlay & is on APKMirror now, so it's definitely official.  Even FfU seems to recognize an error (hence the report above).

**Comment by TPS** [comment]
#369, #370, #371, & #372 seem to be pretty much this same error.

**Comment by eleius** [comment]
> AFAIK there is no bug in FFUpdater. I guess you have to wait for the release of the real 111.0 version.

I have the same issue with Firefox. Official is 111.0 (installed from gplay) but FFUpdater wants to downgrade to 110.1.0

**Comment by Tobi823** [comment]
Shoud be fixed with version 77.7.12
